src/main/java/org/terasology/rendering/nui/market/UIInteractionWrapper.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.nui.market;
+
+import org.terasology.math.geom.Vector2i;
+import org.terasology.rendering.nui.*;
+import org.terasology.rendering.nui.databinding.DefaultBinding;
+import org.terasology.rendering.nui.events.NUIMouseClickEvent;
+import org.terasology.rendering.nui.skin.UISkin;
+import org.terasology.rendering.nui.widgets.ActivateEventListener;
+import org.terasology.rendering.nui.widgets.TooltipLine;
+import org.terasology.rendering.nui.widgets.TooltipLineRenderer;
+import org.terasology.rendering.nui.widgets.UIList;
+import org.terasology.utilities.Assets;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class UIInteractionWrapper extends CoreWidget {
```

Comment:
Should this wrapper be provided by the engine?

Reply:
Yes, we can shift it to engine.